ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/JsonRpcRequestContext.java

```diff
@@ -16,25 +16,38 @@
 
 import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import io.vertx.ext.auth.User;
 
 public class JsonRpcRequestContext {
 
   private final JsonRpcRequest jsonRpcRequest;
   private final Optional<User> user;
+  private final AtomicBoolean alive;
 
   public JsonRpcRequestContext(final JsonRpcRequest jsonRpcRequest) {
-    this(jsonRpcRequest, Optional.empty());
+    this(jsonRpcRequest, new AtomicBoolean(true));
+  }
+
+  public JsonRpcRequestContext(final JsonRpcRequest jsonRpcRequest, final AtomicBoolean alive) {
```

Comment:
Would a `Supplier<Boolean>` also work?  we could pass in a lambda that goes `() -> ctx.response().closed()`  (or `() -> true` for always alive or when we don't have the request context) and not have to worry about keeping another variable state in sync.

Reply:
Changed to a `Supplier<Boolean>`